lonestar/analytics/distributed/tc/tc.cpp

```diff
@@ -115,28 +164,25 @@ int main(int argc, char** argv) {
 #ifdef GALOIS_ENABLE_GPU
   std::tie(hg, syncSubstrate) =
       distGraphInitialization<NodeData, void>(&cuda_ctx, false);
-#else
-  std::tie(hg, syncSubstrate) = distGraphInitialization<NodeData, void>();
-#endif
-
   std::string timer_str("SortEdgesGPU");
   galois::StatTimer edgeSortTime("SortEdgesGPU", regionname);
   edgeSortTime.start();
   sort_cuda(cuda_ctx);
   edgeSortTime.stop();
+#else
+  std::tie(hg, syncSubstrate) = distGraphInitialization<NodeData, void>();
 
```

Comment:
Please expose a function to the user in MiningGraph that they can call to sort the graph by destination similar to how sort_cuda is currently called. Note that if the mirror edge synchronization was required this sort would HAVE to be done in construction itself rather than after construction, but right now since the mirror edge sync isn't being used in any app it should be fine.

i.e. See line 332 in MiningPartitioner.h: make base_DistGraph::sortEdges() something the user can call.

Reply:
"sort" is misleading - please have a more descriptive name for both GPU and CPU functions to sort like sortNodesByDegree

Reply:
In this case we're not actually sorting the nodes by degree: it's just sorting edge destinations.

Introducing the sort by degree functionality that ordered count is using into this code would be a nice idea, but in that case it would have to be done before partitioning occurs or done very cleverly on the partitioned graphs on each host such that that LID-GID handshake doesn't get messed up.

Reply:
....actually, we don't even need LID-GID handshake. 🤔 
The only sync that occurs is DGAccumulator.

This is something we can look into later.

Re: original point, sortEdgesByDest would be a good name.

Reply:
LID-GID mapping is simple to maintain and is not required in this case. 

Each partition can be sort its nodes by degree.